### PR TITLE
Update Truffle to >19.0.1 release

### PR DESCRIPTION
**Pull request by smarr**
This PR gets us onto the latest Truffle version.
There are various little changes, some of which may have bigger impact down the line.

- The interop support is now even more minimal than before, but, it seems sufficient for debugging, and for everything that's currently tested.
- This PR also restore testing on macOS.
- We use now `mx native-image` which simplifies image generation and avoids having to maintain dependencies manually.

@daumayr would be great if you could give this a spin and see whether it works on your machine.

### CODE REVIEW

**Comment by smarr** [comment]
@fniephaus possibly relevant for you, too.

@richard-roberts @kjx @IsaacOscar @Jamigrith: this is a major update of Truffle, and we might want to consider this for https://github.com/gracelang/moth-SOMns It allows to use the officially released GraalVM 19

File: src/som/compiler/MixinDefinition.java [hunk 1]
```diff
@@ -73,6 +72,7 @@
  * at runtime, which then also has the super class and mixins resolved to be
  * used to instantiate {@link SClass} objects.
  */
+@ExportLibrary(InteropLibrary.class)
```
Comment (fniephaus) [inline src/som/compiler/MixinDefinition.java:75]: Why add `@ExportLibrary(InteropLibrary.class)` to this class if you don't implement any of the interop messages (with `@ExportMessage`)?

File: src/som/primitives/threading/TaskThreads.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 public final class TaskThreads {
 
+  @ExportLibrary(InteropLibrary.class)
```
Comment (fniephaus) [inline src/som/primitives/threading/TaskThreads.java:28]: Same question (from above).

File: src/som/vmobjects/SObjectWithClass.java [hunk 3]
```diff
@@ -3,12 +3,13 @@
...
 import com.oracle.truffle.api.frame.MaterializedFrame;
-import com.oracle.truffle.api.interop.ForeignAccess;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
 
-import som.interop.SObjectInteropMessageResolutionForeign;
 import som.interpreter.objectstorage.ClassFactory;
 
 
+@ExportLibrary(InteropLibrary.class)
```
Comment (fniephaus) [inline src/som/vmobjects/SObjectWithClass.java:12]: And again...

File: src/som/compiler/MixinDefinition.java (hunk 1 above)
Comment (smarr) [reply src/som/compiler/MixinDefinition.java:75]: Work in progress

Comment (fniephaus) [reply src/som/compiler/MixinDefinition.java:75]: Ok, I was just wondering because this is not a draft pull request. 😉

Comment (smarr) [reply src/som/compiler/MixinDefinition.java:75]: This is not going to be addressed in this PR. There's no need for more interop support.

**Pull request merged**